### PR TITLE
refactor: Change C style casts to C++ style (Part 5)

### PR DESCRIPTION
**Pull request by aditi-pandit**
As per the security guideline in
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast

Covers the findings in velox/tpch

### CODE REVIEW

**Review by czentgr — APPROVED** [review]
Thanks!

File: velox/tpch/gen/dbgen/include/tpch_constants.hpp [hunk 1]
```diff
@@ -1,9 +1,26 @@
+/*
```
Comment (aditi-pandit) [inline velox/tpch/gen/dbgen/include/tpch_constants.hpp:1]: @pedroerp : I was working with some of the files in the velox/tpch/gen/dbgen folder and there were some inconsistencies in the Facebook header placements. Some files (e.g. file https://github.com/facebookincubator/velox/blob/main/velox/tpch/gen/dbgen/include/dbgen/dss.h) had them, and some not. While working on this PR, as per instructions I did fix the headers I touched, but I wonder if the placement of the copyright is okay from a licensing perspective. 

I am keen to hear your take on it. wdyt ?

File: velox/tpch/gen/dbgen/bm_utils.cpp [hunk 2]
```diff
@@ -398,7 +400,7 @@ char** mk_ascdate(void) {
   dss_time_t t;
   DSS_HUGE i;
 
-  m = (char**)malloc((size_t)(TOTDATE * sizeof(char*)));
+  m = reinterpret_cast<char**>(malloc((size_t)(TOTDATE * sizeof(char*))));
```
Comment (czentgr) [inline velox/tpch/gen/dbgen/bm_utils.cpp:403]: Missed one C-style cast in the malloc.

Comment (aditi-pandit) [reply velox/tpch/gen/dbgen/bm_utils.cpp]: Nice catch. Done.

File: velox/tpch/gen/dbgen/include/tpch_constants.hpp [hunk 3]
```diff
@@ -1,9 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 /* THIS FILE WAS AUTOMATICALLY GENERATED BY generate_csv_header.py */
 
 #pragma once
 
 namespace facebook::velox::tpch::dbgen {
 
+#define CONST_CHAR_CAST reinterpret_cast<const char*>
```
[the remark this answers is not in the document]
Comment (aditi-pandit) [reply velox/tpch/gen/dbgen/include/tpch_constants.hpp:22]: @majetideepak : Curious about this feedback. They are being liberally used in this code-base. 

I had originally written the way you are suggesting... there are many (22 * 4) places and 4 big matrices where this cast occurs, so it ends up being quite unreadable.

File: velox/tpch/gen/dbgen/speed_seed.cpp [hunk 4]
```diff
@@ -105,7 +120,6 @@ void fake_tpch_a_rnd(int min, int max, seed_t* seed) {
 }
 
 long sd_part(int child, DSS_HUGE skip_count, DBGenContext* ctx) {
```
[the remark this answers is not in the document]
Comment (aditi-pandit) [reply velox/tpch/gen/dbgen/speed_seed.cpp]: Nice catch. Done. Thanks.

File: velox/tpch/gen/dbgen/include/tpch_constants.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (aditi-pandit) [reply velox/tpch/gen/dbgen/include/tpch_constants.hpp:22]: @majetideepak That's fair. Adding the TPCH array names makes the pragmas more specific. Though I do have 4 of them now. But that's alright.

Have updated the code.

**Review by majetideepak — APPROVED** [review]
Thanks, @aditi-pandit

File: velox/tpch/gen/dbgen/include/tpch_constants.hpp [hunk 5]
```diff
@@ -1,9 +1,29 @@
...
 /* THIS FILE WAS AUTOMATICALLY GENERATED BY generate_csv_header.py */
 
 #pragma once
 
 namespace facebook::velox::tpch::dbgen {
 
+#define TPCH_QUERY(id) reinterpret_cast<const char*>(TPCH_QUERIES_##id)
+#define TPCH_ANSWER_SF1(id) reinterpret_cast<const char*>(TPCH_ANSWERS_SF1_##id)
+#define TPCH_ANSWER_SF0_1(id) reinterpret_cast<const char*>(TPCH_ANSWERS_SF0_1_##id)
+#define TPCH_ANSWER_SF0_01(id) reinterpret_cast<const char*>(TPCH_ANSWERS_SF0_01_##id)
```
Comment (majetideepak) [inline velox/tpch/gen/dbgen/include/tpch_constants.hpp:25]: @aditi-pandit i noticed you removed the undefs. Let's keep them to avoid leaking the pragmas.

Comment (aditi-pandit) [reply velox/tpch/gen/dbgen/include/tpch_constants.hpp]: Moved the defines and undefs closer to use, so that the scope is restricted to the minimum.

**Comment by majetideepak** [comment]
@bikramSingh91 can you please re-import this? Thanks!

**Comment by bikramSingh91** [comment]
@aditi-pandit 
There are a couple of format issues highlighted by internal linter, posting them here, can you please make the suggested changes? Thanks!

File: `velox/tpch/gen/dbgen/include/tpch_constants.hpp`

lines 1003 - 1012
```

-    TPCH_QUERY(q01), TPCH_QUERY(q02),
-    TPCH_QUERY(q03), TPCH_QUERY(q04),
-    TPCH_QUERY(q05), TPCH_QUERY(q06),
-    TPCH_QUERY(q07), TPCH_QUERY(q08),
-    TPCH_QUERY(q09), TPCH_QUERY(q10),
-    TPCH_QUERY(q11), TPCH_QUERY(q12),
-    TPCH_QUERY(q13), TPCH_QUERY(q14),
-    TPCH_QUERY(q15), TPCH_QUERY(q16),
-    TPCH_QUERY(q17), TPCH_QUERY(q18),
-    TPCH_QUERY(q19), TPCH_QUERY(q20),
+    TPCH_QUERY(q01), TPCH_QUERY(q02), TPCH_QUERY(q03), TPCH_QUERY(q04),
+    TPCH_QUERY(q05), TPCH_QUERY(q06), TPCH_QUERY(q07), TPCH_QUERY(q08),
+    TPCH_QUERY(q09), TPCH_QUERY(q10), TPCH_QUERY(q11), TPCH_QUERY(q12),
+    TPCH_QUERY(q13), TPCH_QUERY(q14), TPCH_QUERY(q15), TPCH_QUERY(q16),
+    TPCH_QUERY(q17), TPCH_QUERY(q18), TPCH_QUERY(q19), TPCH_QUERY(q20),
```

line 2832
```
-#define TPCH_ANSWER_SF0_01(id) reinterpret_cast<const char*>(TPCH_ANSWERS_SF0_01_##id)
+#define TPCH_ANSWER_SF0_01(id) \
+  reinterpret_cast<const char*>(TPCH_ANSWERS_SF0_01_##id)
```

lines 2835 - 2845
```
-    TPCH_ANSWER_SF0_01(q01), TPCH_ANSWER_SF0_01(q02),
-    TPCH_ANSWER_SF0_01(q03), TPCH_ANSWER_SF0_01(q04),
-    TPCH_ANSWER_SF0_01(q05), TPCH_ANSWER_SF0_01(q06),
-    TPCH_ANSWER_SF0_01(q07), TPCH_ANSWER_SF0_01(q08),
-    TPCH_ANSWER_SF0_01(q09), TPCH_ANSWER_SF0_01(q10),
-    TPCH_ANSWER_SF0_01(q11), TPCH_ANSWER_SF0_01(q12),
-    TPCH_ANSWER_SF0_01(q13), TPCH_ANSWER_SF0_01(q14),
-    TPCH_ANSWER_SF0_01(q15), TPCH_ANSWER_SF0_01(q16),
-    TPCH_ANSWER_SF0_01(q17), TPCH_ANSWER_SF0_01(q18),
-    TPCH_ANSWER_SF0_01(q19), TPCH_ANSWER_SF0_01(q20),
-    TPCH_ANSWER_SF0_01(q21), TPCH_ANSWER_SF0_01(q22)};
+    TPCH_ANSWER_SF0_01(q01), TPCH_ANSWER_SF0_01(q02), TPCH_ANSWER_SF0_01(q03),
+    TPCH_ANSWER_SF0_01(q04), TPCH_ANSWER_SF0_01(q05), TPCH_ANSWER_SF0_01(q06),
+    TPCH_ANSWER_SF0_01(q07), TPCH_ANSWER_SF0_01(q08), TPCH_ANSWER_SF0_01(q09),
+    TPCH_ANSWER_SF0_01(q10), TPCH_ANSWER_SF0_01(q11), TPCH_ANSWER_SF0_01(q12),
+    TPCH_ANSWER_SF0_01(q13), TPCH_ANSWER_SF0_01(q14), TPCH_ANSWER_SF0_01(q15),
+    TPCH_ANSWER_SF0_01(q16), TPCH_ANSWER_SF0_01(q17), TPCH_ANSWER_SF0_01(q18),
+    TPCH_ANSWER_SF0_01(q19), TPCH_ANSWER_SF0_01(q20), TPCH_ANSWER_SF0_01(q21),
+    TPCH_ANSWER_SF0_01(q22)};
```

line 13430
```
-#define TPCH_ANSWER_SF0_1(id) reinterpret_cast<const char*>(TPCH_ANSWERS_SF0_1_##id)
+#define TPCH_ANSWER_SF0_1(id) \
+  reinterpret_cast<const char*>(TPCH_ANSWERS_SF0_1_##id)
```


lines 13433 - 13443
```
-    TPCH_ANSWER_SF0_1(q01), TPCH_ANSWER_SF0_1(q02),
-    TPCH_ANSWER_SF0_1(q03), TPCH_ANSWER_SF0_1(q04),
-    TPCH_ANSWER_SF0_1(q05), TPCH_ANSWER_SF0_1(q06),
-    TPCH_ANSWER_SF0_1(q07), TPCH_ANSWER_SF0_1(q08),
-    TPCH_ANSWER_SF0_1(q09), TPCH_ANSWER_SF0_1(q10),
-    TPCH_ANSWER_SF0_1(q11), TPCH_ANSWER_SF0_1(q12),
-    TPCH_ANSWER_SF0_1(q13), TPCH_ANSWER_SF0_1(q14),
-    TPCH_ANSWER_SF0_1(q15), TPCH_ANSWER_SF0_1(q16),
-    TPCH_ANSWER_SF0_1(q17), TPCH_ANSWER_SF0_1(q18),
-    TPCH_ANSWER_SF0_1(q19), TPCH_ANSWER_SF0_1(q20),
-    TPCH_ANSWER_SF0_1(q21), TPCH_ANSWER_SF0_1(q22)};
+    TPCH_ANSWER_SF0_1(q01), TPCH_ANSWER_SF0_1(q02), TPCH_ANSWER_SF0_1(q03),
+    TPCH_ANSWER_SF0_1(q04), TPCH_ANSWER_SF0_1(q05), TPCH_ANSWER_SF0_1(q06),
+    TPCH_ANSWER_SF0_1(q07), TPCH_ANSWER_SF0_1(q08), TPCH_ANSWER_SF0_1(q09),
+    TPCH_ANSWER_SF0_1(q10), TPCH_ANSWER_SF0_1(q11), TPCH_ANSWER_SF0_1(q12),
+    TPCH_ANSWER_SF0_1(q13), TPCH_ANSWER_SF0_1(q14), TPCH_ANSWER_SF0_1(q15),
+    TPCH_ANSWER_SF0_1(q16), TPCH_ANSWER_SF0_1(q17), TPCH_ANSWER_SF0_1(q18),
+    TPCH_ANSWER_SF0_1(q19), TPCH_ANSWER_SF0_1(q20), TPCH_ANSWER_SF0_1(q21),
+    TPCH_ANSWER_SF0_1(q22)};
```

lines 60731 - 60741
```
-    TPCH_ANSWER_SF1(q01), TPCH_ANSWER_SF1(q02),
-    TPCH_ANSWER_SF1(q03), TPCH_ANSWER_SF1(q04),
-    TPCH_ANSWER_SF1(q05), TPCH_ANSWER_SF1(q06),
-    TPCH_ANSWER_SF1(q07), TPCH_ANSWER_SF1(q08),
-    TPCH_ANSWER_SF1(q09), TPCH_ANSWER_SF1(q10),
-    TPCH_ANSWER_SF1(q11), TPCH_ANSWER_SF1(q12),
-    TPCH_ANSWER_SF1(q13), TPCH_ANSWER_SF1(q14),
-    TPCH_ANSWER_SF1(q15), TPCH_ANSWER_SF1(q16),
-    TPCH_ANSWER_SF1(q17), TPCH_ANSWER_SF1(q18),
-    TPCH_ANSWER_SF1(q19), TPCH_ANSWER_SF1(q20),
-    TPCH_ANSWER_SF1(q21), TPCH_ANSWER_SF1(q22)};
+    TPCH_ANSWER_SF1(q01), TPCH_ANSWER_SF1(q02), TPCH_ANSWER_SF1(q03),
+    TPCH_ANSWER_SF1(q04), TPCH_ANSWER_SF1(q05), TPCH_ANSWER_SF1(q06),
+    TPCH_ANSWER_SF1(q07), TPCH_ANSWER_SF1(q08), TPCH_ANSWER_SF1(q09),
+    TPCH_ANSWER_SF1(q10), TPCH_ANSWER_SF1(q11), TPCH_ANSWER_SF1(q12),
+    TPCH_ANSWER_SF1(q13), TPCH_ANSWER_SF1(q14), TPCH_ANSWER_SF1(q15),
+    TPCH_ANSWER_SF1(q16), TPCH_ANSWER_SF1(q17), TPCH_ANSWER_SF1(q18),
+    TPCH_ANSWER_SF1(q19), TPCH_ANSWER_SF1(q20), TPCH_ANSWER_SF1(q21),
+    TPCH_ANSWER_SF1(q22)};
```

**Comment by aditi-pandit** [comment]
Thanks @bikramSingh91. Have made the changes. PTAL.

**Comment by bikramSingh91** [comment]
@aditi-pandit Unfortunately after you fixed the previous format changes, some more popped up. My guess is that format rules for macros is a bit wonky. Pasting them here again, hope it'll be the last of it.

file: `fbcode/velox/tpch/gen/dbgen/include/tpch_constants.hpp`

line 2827
```
-#define TPCH_ANSWER_SF0_01(id) reinterpret_cast<const char*>(TPCH_ANSWERS_SF0_01_##id)
+#define TPCH_ANSWER_SF0_01(id) \
+  reinterpret_cast<const char*>(TPCH_ANSWERS_SF0_01_##id)
```

line 2830 - 2835
```
-    TPCH_ANSWER_SF0_01(q01), TPCH_ANSWER_SF0_01(q02), TPCH_ANSWER_SF0_01(q03), TPCH_ANSWER_SF0_01(q04),
-    TPCH_ANSWER_SF0_01(q05), TPCH_ANSWER_SF0_01(q06), TPCH_ANSWER_SF0_01(q07), TPCH_ANSWER_SF0_01(q08),
-    TPCH_ANSWER_SF0_01(q09), TPCH_ANSWER_SF0_01(q10), TPCH_ANSWER_SF0_01(q11), TPCH_ANSWER_SF0_01(q12),
-    TPCH_ANSWER_SF0_01(q13), TPCH_ANSWER_SF0_01(q14), TPCH_ANSWER_SF0_01(q15), TPCH_ANSWER_SF0_01(q16),
-    TPCH_ANSWER_SF0_01(q17), TPCH_ANSWER_SF0_01(q18), TPCH_ANSWER_SF0_01(q19), TPCH_ANSWER_SF0_01(q20),
-    TPCH_ANSWER_SF0_01(q21), TPCH_ANSWER_SF0_01(q22)};
+    TPCH_ANSWER_SF0_01(q01), TPCH_ANSWER_SF0_01(q02), TPCH_ANSWER_SF0_01(q03),
+    TPCH_ANSWER_SF0_01(q04), TPCH_ANSWER_SF0_01(q05), TPCH_ANSWER_SF0_01(q06),
+    TPCH_ANSWER_SF0_01(q07), TPCH_ANSWER_SF0_01(q08), TPCH_ANSWER_SF0_01(q09),
+    TPCH_ANSWER_SF0_01(q10), TPCH_ANSWER_SF0_01(q11), TPCH_ANSWER_SF0_01(q12),
+    TPCH_ANSWER_SF0_01(q13), TPCH_ANSWER_SF0_01(q14), TPCH_ANSWER_SF0_01(q15),
+    TPCH_ANSWER_SF0_01(q16), TPCH_ANSWER_SF0_01(q17), TPCH_ANSWER_SF0_01(q18),
+    TPCH_ANSWER_SF0_01(q19), TPCH_ANSWER_SF0_01(q20), TPCH_ANSWER_SF0_01(q21),
+    TPCH_ANSWER_SF0_01(q22)};
```

line 13420
```
-#define TPCH_ANSWER_SF0_1(id) reinterpret_cast<const char*>(TPCH_ANSWERS_SF0_1_##id)
+#define TPCH_ANSWER_SF0_1(id) \
+  reinterpret_cast<const char*>(TPCH_ANSWERS_SF0_1_##id)
```

line 13423 - 13428
```
-    TPCH_ANSWER_SF0_1(q01), TPCH_ANSWER_SF0_1(q02), TPCH_ANSWER_SF0_1(q03), TPCH_ANSWER_SF0_1(q04),
-    TPCH_ANSWER_SF0_1(q05), TPCH_ANSWER_SF0_1(q06), TPCH_ANSWER_SF0_1(q07), TPCH_ANSWER_SF0_1(q08),
-    TPCH_ANSWER_SF0_1(q09), TPCH_ANSWER_SF0_1(q10), TPCH_ANSWER_SF0_1(q11), TPCH_ANSWER_SF0_1(q12),
-    TPCH_ANSWER_SF0_1(q13), TPCH_ANSWER_SF0_1(q14), TPCH_ANSWER_SF0_1(q15), TPCH_ANSWER_SF0_1(q16),
-    TPCH_ANSWER_SF0_1(q17), TPCH_ANSWER_SF0_1(q18), TPCH_ANSWER_SF0_1(q19), TPCH_ANSWER_SF0_1(q20),
-    TPCH_ANSWER_SF0_1(q21), TPCH_ANSWER_SF0_1(q22)};
+    TPCH_ANSWER_SF0_1(q01), TPCH_ANSWER_SF0_1(q02), TPCH_ANSWER_SF0_1(q03),
+    TPCH_ANSWER_SF0_1(q04), TPCH_ANSWER_SF0_1(q05), TPCH_ANSWER_SF0_1(q06),
+    TPCH_ANSWER_SF0_1(q07), TPCH_ANSWER_SF0_1(q08), TPCH_ANSWER_SF0_1(q09),
+    TPCH_ANSWER_SF0_1(q10), TPCH_ANSWER_SF0_1(q11), TPCH_ANSWER_SF0_1(q12),
+    TPCH_ANSWER_SF0_1(q13), TPCH_ANSWER_SF0_1(q14), TPCH_ANSWER_SF0_1(q15),
+    TPCH_ANSWER_SF0_1(q16), TPCH_ANSWER_SF0_1(q17), TPCH_ANSWER_SF0_1(q18),
+    TPCH_ANSWER_SF0_1(q19), TPCH_ANSWER_SF0_1(q20), TPCH_ANSWER_SF0_1(q21),
+    TPCH_ANSWER_SF0_1(q22)};
```

line 60716 - 60721
```
-    TPCH_ANSWER_SF1(q01), TPCH_ANSWER_SF1(q02), TPCH_ANSWER_SF1(q03), TPCH_ANSWER_SF1(q04),
-    TPCH_ANSWER_SF1(q05), TPCH_ANSWER_SF1(q06), TPCH_ANSWER_SF1(q07), TPCH_ANSWER_SF1(q08),
-    TPCH_ANSWER_SF1(q09), TPCH_ANSWER_SF1(q10), TPCH_ANSWER_SF1(q11), TPCH_ANSWER_SF1(q12),
-    TPCH_ANSWER_SF1(q13), TPCH_ANSWER_SF1(q14), TPCH_ANSWER_SF1(q15), TPCH_ANSWER_SF1(q16),
-    TPCH_ANSWER_SF1(q17), TPCH_ANSWER_SF1(q18), TPCH_ANSWER_SF1(q19), TPCH_ANSWER_SF1(q20),
-    TPCH_ANSWER_SF1(q21), TPCH_ANSWER_SF1(q22)};
+    TPCH_ANSWER_SF1(q01), TPCH_ANSWER_SF1(q02), TPCH_ANSWER_SF1(q03),
+    TPCH_ANSWER_SF1(q04), TPCH_ANSWER_SF1(q05), TPCH_ANSWER_SF1(q06),
+    TPCH_ANSWER_SF1(q07), TPCH_ANSWER_SF1(q08), TPCH_ANSWER_SF1(q09),
+    TPCH_ANSWER_SF1(q10), TPCH_ANSWER_SF1(q11), TPCH_ANSWER_SF1(q12),
+    TPCH_ANSWER_SF1(q13), TPCH_ANSWER_SF1(q14), TPCH_ANSWER_SF1(q15),
+    TPCH_ANSWER_SF1(q16), TPCH_ANSWER_SF1(q17), TPCH_ANSWER_SF1(q18),
+    TPCH_ANSWER_SF1(q19), TPCH_ANSWER_SF1(q20), TPCH_ANSWER_SF1(q21),
+    TPCH_ANSWER_SF1(q22)};
```

**Comment by aditi-pandit** [comment]
@bikramSingh91 : Did the next round of linter fixes :( Hope this resolves the issues. Thanks for your patience with this.

**Comment by bikramSingh91** [comment]
One more popped up, this would be the last for sure!
I realized a bit late that I could just submit local patches internally to re-trigger the checks. So eventually did that for this last linter error, and this seems to clear all. Thanks for tagging along with this stream of linter updates. Appreciate your patience!

lines 60719 - 60720
```
-#define TPCH_ANSWER_SF1(id) \
-  reinterpret_cast<const char*>(TPCH_ANSWERS_SF1_##id)
+#define TPCH_ANSWER_SF1(id) reinterpret_cast<const char*>(TPCH_ANSWERS_SF1_##id)
```

**Comment by aditi-pandit** [comment]
@bikramSingh91 : Sounds like you don't need the linter fix from me. I was thinking that I will push a commit in any case, but don't want to cause further complications in the merge process.

Definitely ping me if you need anything else. I've sent you a slack as well.

**Comment by aditi-pandit** [comment]
Thanks @bikramSingh91 for your message.

Have updated the code accordingly.

**Pull request closed without merging**